### PR TITLE
infra: [poc] Add pre-commit linting hook

### PR DESCRIPTION
**Pull request by RubenRBS**
The basics look pretty good, but there are some issues given how CCI is setup:

 - The hooks are not automatically installed, as we don't ask contributors to install a `requirements-dev.txt` nor have a `setup.py` file, so the user needs to know to run these commands to get the hooks working:
```
pip install pre-commit
pre-commit install
set PYTHONPATH=/whatever/path/to/cci
```
 - Right now warnings are not logged to the console at any point other than if an error was also found, so things like the "Consider using f-strings" lint rule do not show up unless the user manually runs the linter. I _think_ this can be changed, but I don't know if it makes much sense either

As you can see, my commit now takes 5s to complete (up from the less than 1 with `--no-verify`), which I think it's a reasonable tradeoff
image

Happy to discuss where to take this next

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+cfgv==3.3.1
```
Comment (prince-chrismc) [inline requirements.txt:1]: Conan less then 2 maybe as well?

**Comment by prince-chrismc** [comment]
I dont think people will understand/appreciate the benefit of this since they dont know how it currently works

I would suggest replace the "Running X Linter" sections from https://github.com/conan-io/conan-center-index/blob/master/docs/developing_recipes_locally.md#contents with a "Using pre-commit hooks" along side the regular conan center hooks

I think that would be a very easy way to show how much easier this is

File: requirements.txt (hunk 1 above)
Comment (RubenRBS) [reply requirements.txt:1]: Yeah this is something I wanted to discuss in the next Shepherds :)

Comment (RubenRBS) [reply requirements.txt:1]: For now, let's not have Conan present at first, and if we see the need to have it, we can add it at a later date

**Comment by RubenRBS** [comment]
Not stale, just waiting

**Pull request closed without merging**